index.js

```diff
@@ -155,3 +176,45 @@ function unproject(z, x, y) {
     var lat = 360 / Math.PI * Math.atan(Math.exp((180 - y * 360 / z2) * Math.PI / 180)) - 90;
     return [lng, lat];
 }
+
+function updateInfo(mbtiles, name, info, callback) {
+    mbtiles._db.get(
+        'SELECT MAX(tile_column) AS maxx, ' +
+        'MIN(tile_column) AS minx, MAX(tile_row) AS maxy, ' +
+        'MIN(tile_row) AS miny FROM tiles ' +
+        'WHERE zoom_level = ?',
+        info.minzoom,
+        function(err, row) {
+            if (err) return callback(err);
+            if (!row) return callback(null, info);
+
+            // @TODO this breaks a little at zoom level zero
+            var urTile = sm.bbox(row.maxx, row.maxy, info.minzoom, true);
+            var llTile = sm.bbox(row.minx, row.miny, info.minzoom, true);
+            // @TODO bounds are limited to "sensible" values here
+            // as sometimes tilesets are rendered with "negative"
+            // and/or other extremity tiles. Revisit this if there
+            // are actual use cases for out-of-bounds bounds.
+
+            info.bounds = [
+                llTile[0] > -180 ? llTile[0] : -180,
+                llTile[1] > -90 ? llTile[1] : -90,
+                urTile[2] < 180 ? urTile[2] : 180,
+                urTile[3] < 90 ? urTile[3] : 90
+            ];
+
+            var range = info.maxzoom - info.minzoom;
+            info.center = [
+                (info.bounds[2] - info.bounds[0]) / 2 + info.bounds[0],
+                (info.bounds[3] - info.bounds[1]) / 2 + info.bounds[1],
+                range <= 1 ? info.maxzoom : Math.floor(range * 0.5) + info.minzoom
+            ];
+            info.name = info.description = info.basename = name;
+            info.vector_layers[0].fields = {};
+            mbtiles.putInfo(info, function (err) {
+                if (err) throw err;
+
+                return callback(null, mbtiles);
+            });
+        });
```

Comment:
Indentation is a bit awkward here? I'd put the `)` on the next line to close `db.get( .. `